gateway/src/main/java/io/pravega/example/iot/gateway/DataHandler.java

```diff
@@ -83,4 +93,41 @@ public String postData(@Context Request request, String data, @PathParam("remote
             throw e;
         }
     }
+
+    @POST
+    @Path("/rawData/{remoteAddr}")
+    public String postRawData(@Context Request request, byte[] data, @PathParam("remoteAddr") String remoteAddr) throws Exception {
+        final long ingestTimestamp = System.currentTimeMillis();
+        final String ingestTimestampStr = dateFormat.format(new Date(ingestTimestamp));
+        try {
+            if(data == null)
+                return "{DATA Not Received}";
```

Comment:
Will this method ever receive a null value for data? If the HTTP body is empty, won't this send an empty byte[]? If a null value is not a possible input, then there is no reason to test for it. Even if it does get a null, this method will just throw a NPE.

Reply:
good point. removed the check.